data-prepper-plugins/common/src/test/java/com/amazon/dataprepper/plugins/source/file/FileSourceTests.java

```diff
@@ -0,0 +1,248 @@
+
+    @Mock
+    private PluginFactory pluginFactory;
+
+    private Buffer<Record<Object>> buffer;
+
+    private Map<String, Object> pluginSettings;
+
+    private List<Record<Object>> expectedEventsPlain;
+    private List<Record<Object>> expectedEventsJson;
+    private List<Record<Object>> expectedEventsInvalidJson;
+
+
+    @BeforeEach
+    public void setup() {
+        pluginSettings = new HashMap<>();
+        expectedEventsPlain = new ArrayList<>();
+        expectedEventsJson = new ArrayList<>();
+        expectedEventsInvalidJson = new ArrayList<>();
+
+        pluginSettings.put(FileSourceConfig.ATTRIBUTE_PATH, TEST_FILE_PATH_PLAIN);
+
+        // plain
+        final String expectedPlainFirstLine = "THIS IS A PLAINTEXT LINE";
+        final String expectedPlainSecondLine = "THIS IS ANOTHER PLAINTEXT LINE";
+
+
+        final Record<Object> firstEventPlain = createRecordEventWithKeyValuePair(FileSource.MESSAGE_KEY, expectedPlainFirstLine);
+        final Record<Object> secondEventPlain = createRecordEventWithKeyValuePair(FileSource.MESSAGE_KEY, expectedPlainSecondLine);
+
+        expectedEventsPlain.add(firstEventPlain);
+        expectedEventsPlain.add(secondEventPlain);
+
+        //json
+        final String expectedJsonFirstLine = "{\"test_key: \"test_value\"}";
+        final String expectedJsonSecondLine = "{\"second_test_key\": \"second_test_value\"}";
+
+        final Record<Object> firstEventJson = createRecordEventWithKeyValuePair("test_key", "test_value");
+        final Record<Object> secondEventJson = createRecordEventWithKeyValuePair("second_test_key", "second_test_value");
+
+        expectedEventsJson.add(firstEventJson);
+        expectedEventsJson.add(secondEventJson);
+
+        // invalid json
+        final String expectedInvalidJsonFirstLine = "{\"test_key: test_value\"}";
+        final String expectedInvalidJsonSecondLine = "{\"second_test_key\": \"second_test_value\"";
+
+
+        final Record<Object> firstEventInvalidJson = createRecordEventWithKeyValuePair(FileSource.MESSAGE_KEY, expectedInvalidJsonFirstLine);
+        final Record<Object> secondEventInvalidJson = createRecordEventWithKeyValuePair(FileSource.MESSAGE_KEY, expectedInvalidJsonSecondLine);
+
+        expectedEventsInvalidJson.add(firstEventInvalidJson);
+        expectedEventsInvalidJson.add(secondEventInvalidJson);
+
+
+
+        buffer = getBuffer();
+    }
+
+    private FileSource createObjectUnderTest() {
+        fileSourceConfig = OBJECT_MAPPER.convertValue(pluginSettings, FileSourceConfig.class);
+        return new FileSource(fileSourceConfig, pluginMetrics, pluginFactory);
+    }
+
+    private BlockingBuffer<Record<Object>> getBuffer() {
+        final HashMap<String, Object> integerHashMap = new HashMap<>();
+        integerHashMap.put("buffer_size", 2);
+        integerHashMap.put("batch_size", 2);
+        final PluginSetting pluginSetting = new PluginSetting("blocking_buffer", integerHashMap) {{
```

Comment:
This is an anti-pattern which I'm hoping to get away from in Data Prepper. This is creating an anonymous class with a static initializer. It is better to to break this into two statements - one for `new` and the other to `setPipelineName`.

You can probably even create a mock which is better.